lib/Disco.php

```diff
@@ -104,13 +117,13 @@ public function __construct(
 
             if ($state !== null) {
                 if (isset($state[self::SAML_REQUESTED_AUTHN_CONTEXT][self::AUTHN_CONTEXT_CLASS_REF])) {
-                    $this->authnContextClassRef = $state[self::SAML_REQUESTED_AUTHN_CONTEXT]
+                    $this->originalAuthnContextClassRef = $state[self::SAML_REQUESTED_AUTHN_CONTEXT]
                         [self::AUTHN_CONTEXT_CLASS_REF];
                     $this->removeAuthContextClassRefWithPrefix($state);
                 }
 
                 $id = State::saveState($state, self::SAML_SP_SSO);
-
+                $this->stateId = $id;
```

Comment:
State can be stored directly to $this->stateId

Reply:
And this param is set only if state is created and I don't tell if it is correct